### PR TITLE
Conditional creation of TGW route tables for Central VPCs

### PR DESCRIPTION
**Pull request by alexeygumirov**
In my project we are using your module for creation of central Inspection VPC, as well ass central Ingress and Egress VPCs. But the TGW we create using our own module (we have quite complex logic there), as well as  association and propagation of VPC attachments to our TGW we maintain with our own modules.
Hence, we don't need for the hubandspoke module to create TGW Route Tables and make association and propagation for the Central VPCs.
I have added this capability in the module with the `associate_and_propagate_to_tgw` boolean parameter for Central VPCs. 
This modification is backward compatible.
When this parameter is not provided - it is implicitly set to `true`.
If TGW is created by the hubandspoke module - it always `true`, even if set to `false` (I assume that if someone wants to created TGW with the hubandspoke module, then he trusts it to do the routing in TGW as well).

I did your tests with the modified module as well as I wrote two new tests to check that option works correctly.

### CODE REVIEW

**Comment by drewmullen** [comment]
Hi @alexeygumirov thanks for taking the time to submit this PR. Give us some time to review. If possible can you join the slack channel [Cloud networking as Code](https://join.slack.com/t/cloud-network-as-code/shared_invite/zt-24l84zq11-kUZBsa21~vaY7~Lg9uWwJw)?

**Comment by alexeygumirov** [comment]
Hi guys. Did you have a change to look into it?

File: main.tf [hunk 1]
```diff
@@ -165,7 +169,7 @@ resource "aws_ec2_transit_gateway_prefix_list_reference" "spokes_to_inspection_n
 # Static Route (0.0.0.0/0) from Inspection VPC to Egress VPC if:
 # 1/ Both Inspection VPC and Egress VPC are created, and the traffic inspection is "all" or "north-south".
 resource "aws_ec2_transit_gateway_route" "inspection_to_egress_default_route" {
-  count = local.inspection_and_egress_routes ? 1 : 0
+  count = local.inspection_and_egress_routes && try(local.associate_and_propagate_to_tgw["inspection"], true) ? 1 : 0
```
Comment (drewmullen) [inline main.tf:172]: nitpick for **each of these changes** to meta arguments:

```suggestion
  count = (
    local.inspection_and_egress_routes && 
    try(local.associate_and_propagate_to_tgw["inspection"], true)
  ) ? 1 : 0
```

File: test/central_egress_ingress_no_tgw_association_propagation/test_helpers/providers.tf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      source  = "hashicorp/aws"
+      version = ">= 3.73.0"
+    }
+  }
+}
+
+provider "aws" {
+  default_tags {
+    tags = {
+      "owner" = "Terratest"
```
Comment (drewmullen) [inline test/central_egress_ingress_no_tgw_association_propagation/test_helpers/providers.tf:14]: nice touch!

File: test/central_egress_ingress_no_tgw_association_propagation/central_egress_ingress_no_tgw_association_propagation_test.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    "github.com/stretchr/testify/assert"
+
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+    "github.com/gruntwork-io/terratest/modules/random"
+)
+
+func TestCentralEgressIngressNoAssociationAndPropagation(t *testing.T) {
+    helperDir := "./test_helpers"
```
Comment (drewmullen) [inline test/central_egress_ingress_no_tgw_association_propagation/central_egress_ingress_no_tgw_association_propagation_test.go:14]: for consistency across the codebase can you please rename this dir to `hcl_fixtures`

File: test/central_shared_services_no_tgw_association_propagation/go.mod [hunk 4]
```diff
@@ -0,0 +1,59 @@
+module shared-services-test
```
Comment (drewmullen) [inline test/central_shared_services_no_tgw_association_propagation/go.mod:1]: Please remove any go.mod or go.sum files. I know its frustrating but we do not keep *any* go mod or sum files in the code base. They should be included in the `.gitignore` also. The reasoning comes from security alerting in the dependency chain.

I realize that may seem unintuitive but that is our current requirement

Not a requirement for this PR but a FYI, we are going to be transitioning away from `terratest` to `terraform test` after a few more patch versions come out. We've already begun the process in the VPC module: https://github.com/aws-ia/terraform-aws-vpc/pull/127/files

**Review by drewmullen — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (alexeygumirov) [reply main.tf]: Fixed. All statements like that are better formatted now.

File: test/central_egress_ingress_no_tgw_association_propagation/central_egress_ingress_no_tgw_association_propagation_test.go (hunk 3 above)
Comment (alexeygumirov) [reply test/central_egress_ingress_no_tgw_association_propagation/central_egress_ingress_no_tgw_association_propagation_test.go]: Done. In all my tests I renamed folders and relevant variables.

File: test/central_shared_services_no_tgw_association_propagation/go.mod (hunk 4 above)
Comment (alexeygumirov) [reply test/central_shared_services_no_tgw_association_propagation/go.mod]: .gitignore is updated and go.mod and go.sum files are removed.

**Review by drewmullen — APPROVED** [review]
This LGTM! Thank you very much @alexeygumirov

**Review by pablo19sc — APPROVED** [review]
Re-tested everything - all good.

Thanks for the contribution!

**Review by tbulding — APPROVED** [review]

**Pull request merged**